pkg/k8s/labels.go

```diff
@@ -106,6 +109,15 @@ func GetControllerNs(objectMeta meta.ObjectMeta) string {
 	return objectMeta.Labels[ControllerNSLabel]
 }
 
+func GetOwnerType(objectMeta meta.ObjectMeta) string {
+	for _, label := range podOwnerLabels {
+		if _, ok := objectMeta.Labels[label]; ok {
+			return toOwnerLabel(label)
```

Comment:
Note that this sends back the following labels for the owner segment of the identity string:

* deployment
* replication_controller
* replica_set
* k8s_job
* daemon_set
* stateful_set

That matches the `dst_*` labels that are exported to prometheus, but the underscores might be problematic. If we need those segments to be represented differently, I suggest we try to tackle #740 to fix it.

Reply:
Right now the DNS label validator in the proxy accepts underscores in DNS labels, even though various DNS specs say that they are not legal. I would like to use only 100% valid DNS labels, but that's really an aesthetic thing that's relatively unimportant, so we can/should address #740 asynchronously with this.

Reply:
Aesthetically--and this is totally not a blocker--I think it would be nice to shorten these to their cannonical short representation, i.e.: deploy, rc, rs, job, ds, ss, po...

Reply:
This is good feedback, and I agree that the short form for the owner type would be better in the long run. I'd like to proceed with this implementation for now, however. At the moment we have a single code path for deriving a pod owner string from a `k8s.io/api/core/v1.Pod` object. I'm worried that if we start introducing multiple different strings representing the same pod owner, we'll have trouble keeping track of which type of string representation we're expecting. I think we need a more holistic approach to addressing that, which is covered in #740 (and related to #741).